interpreter/CMakeLists.txt

```diff
@@ -134,24 +160,243 @@ set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES "")
+    mark_as_advanced(LLVM_ENABLE_ASSERTIONS)
+  endif()
+
+  set(LLVM_TOOLS_BINARY_DIR ${TOOLS_BINARY_DIR} CACHE PATH "Path to llvm/bin")
+  set(LLVM_LIBRARY_DIR ${LIBRARY_DIR} CACHE PATH "Path to llvm/lib")
+  set(LLVM_MAIN_INCLUDE_DIR ${INCLUDE_DIR} CACHE PATH "Path to llvm/include")
+  set(LLVM_BINARY_DIR ${LLVM_OBJ_ROOT} CACHE PATH "Path to LLVM build tree")
+  set(LLVM_MAIN_SRC_DIR ${MAIN_SRC_DIR} CACHE PATH "Path to LLVM source tree")
+
+  # Normalize LLVM_CMAKE_PATH. --cmakedir might contain backslashes.
+  # CMake assumes slashes as PATH.
+  file(TO_CMAKE_PATH ${LLVM_CONFIG_CMAKE_PATH} LLVM_CMAKE_PATH)
+
+  find_program(LLVM_TABLEGEN_EXE "llvm-tblgen" ${LLVM_TOOLS_BINARY_DIR}
+    NO_DEFAULT_PATH)
+
+  set(LLVMCONFIG_FILE "${LLVM_CMAKE_PATH}/LLVMConfig.cmake")
+  if(EXISTS ${LLVMCONFIG_FILE})
+    list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_PATH}")
+    include(${LLVMCONFIG_FILE})
+  else()
+    message(FATAL_ERROR "Not found: ${LLVMCONFIG_FILE}")
+  endif()
+
+  # They are used as destination of target generators.
+  # We try to keep these locations consistent with the builtin_llvm. This is important
+  # for the LLVMRES target.
+  # FIXME: In longer term, we do not really need this and may want to adjust LLVMRES.
+  set(LLVM_RUNTIME_OUTPUT_INTDIR ${CMAKE_CURRENT_BINARY_DIR}/llvm/src/${CMAKE_CFG_INTDIR}/bin)
+  set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_CURRENT_BINARY_DIR}/llvm/src/${CMAKE_CFG_INTDIR}/lib${LLVM_LIBDIR_SUFFIX})
+
+  if(WIN32 OR CYGWIN)
+    # DLL platform -- put DLLs into bin.
+    set(LLVM_SHLIB_OUTPUT_INTDIR ${LLVM_RUNTIME_OUTPUT_INTDIR})
+  else()
+    set(LLVM_SHLIB_OUTPUT_INTDIR ${LLVM_LIBRARY_OUTPUT_INTDIR})
+  endif()
+
+  option(LLVM_ENABLE_WARNINGS "Enable compiler warnings." ON)
+  option(LLVM_INSTALL_TOOLCHAIN_ONLY
+    "Only include toolchain files in the 'install' target." OFF)
 
-#---Make LLVM compilation flags public---------------------------------------------------------------
-# Extract the compilation flags from LLVM and make them public to the
-# rest of ROOT so that we can compile against LLVM with matching flags.
+  option(LLVM_FORCE_USE_OLD_HOST_TOOLCHAIN
+    "Set to ON to force using an old, unsupported host toolchain." OFF)
+  option(CLANG_ENABLE_BOOTSTRAP "Generate the clang bootstrap target" OFF)
 
-# LLVM doesn't really give us a API to get this with an in-source build
-# so we just use the normal way of doing this and read the llvm directory
-# compilation properties.
-get_directory_property(LLVM_DEFS DIRECTORY llvm/src COMPILE_DEFINITIONS)
-# Turns DEFINE1;DEFINE2 to -DDEFINE1 -DDEFINE2
-string (REPLACE ";" " -D" LLVM_DEFS ";${LLVM_DEFS}")
-# CACHE variable that contains the additional flags that LLVM needs.
-set(ROOT_LLVM_FLAGS "${LLVM_DEFS}" CACHE STRING "" FORCE)
+  include(AddLLVM)
+  include(TableGen)
+  include(HandleLLVMOptions)
+  include(VersionFromVCS)
+
+  set(PACKAGE_VERSION "${LLVM_PACKAGE_VERSION}")
+  if (NOT PACKAGE_VERSION VERSION_EQUAL 5.0)
```

Comment:
Your PR says it also works with 5.0.1, is this true here if `PACKAGE_VERSION` is 5.0.1?

Reply:
Yes, that is correct. LLVM 5.0.1 is a patch release and is supposed to be ABI compatible (to my surprise ;). On systems such as Ubuntu if you install llvm-5.0-dev you get 5.0.1 actually, so it has to work with 5.0.1.

Reply:
No, what I mean is if `PACKAGE_VERSION VERSION_EQUAL 5.0` evaluates to true even is the version is `5.0.1` and not `5.0`. I guess it does then?

Reply:
Yes, that's the intent.